### DISCUSSION TITLE
404 when using app router

### DISCUSSION THREAD

**Discussion by RuellePaul** [vercel/next.js#54227] [Help]
### Summary

Hello, i'm currently migrating to new app router and i'm facing 404s when defining simple route.

✅ / is good and show home page
❌ /app leads to 404

This is the structure of my app

![Capture d’écran 2023-08-18 à 16 32 44](https://github.com/vercel/next.js/assets/47316542/3a324c27-bbe0-498c-bc0f-4f5bbdc87ef0)

and this is `app.tsx` file :

```
export default function AppView() {
	return <div>App</div>;
}
```

when I navigate to '/app', I see my layout wrapping the default 404 page.

Any clue why ?

### Additional information

using next@13.4.18

**Comment by SayedEnam** [comment]
**Here are some steps you can take to troubleshoot and resolve the "404 error" when using the Next.js app router:**

Check Pages Structure: In Next.js, each file in the pages directory corresponds to a route. Make sure that your pages are structured correctly. For example, if you have a file named about.js in the pages directory, it should be accessible at the /about route.

Dynamic Routes: If you are using dynamic routes (routes with parameters), ensure that your dynamic route files are named correctly and are located in the pages directory. For example, a dynamic route like /users/[id].js should be accessible at a URL like /users/123.

Custom Server: If you're using a custom server with Next.js, ensure that the server is properly configured to handle routing. The server should be set up to fallback to the Next.js routing mechanism for routes that are not explicitly handled.

Static Files: If you're serving static files using Next.js, make sure that the file paths and routes are correctly set up. Static files are typically placed in the public directory and accessed directly.

Link Component: When navigating between pages within your Next.js application, use the Link component provided by Next.js. This helps ensure that client-side navigation is handled properly.

getServerSideProps or getStaticProps: If you are using server-side rendering (SSR) with Next.js, make sure that your getServerSideProps or getStaticProps functions are correctly implemented and not causing any errors that could result in a "404 error."

Check Console and Network Tab: Open your browser's developer console and look for any error messages related to the failed request. Check the Network tab to see the HTTP status code and response details.

Router Configuration: If you're using a custom routing library within your Next.js application, make sure that it's properly integrated and configured.

Check Route Paths: Double-check that you're using the correct route paths when using the Link component or programmatically navigating between pages.

Deployment Configuration: If the issue occurs after deploying your Next.js application, make sure that the deployment process is correctly set up and that the routes are being handled as expected on the deployment platform.
